packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -70,6 +69,18 @@ interface AutoSize extends BaseProps {
   children: ReactNode;
 }
 
+interface Collapsible {
+  collapsible: boolean;
+  hoverHeader?: never;
+  hoverHeaderOffset?: never;
+}
+
+interface HoverHeader {
+  collapsible?: never;
+  hoverHeader?: boolean;
+  hoverHeaderOffset?: number;
+}
```

Comment:
for slightly easier ergonomics, and given (according to the linked issue) a panel can either be in `default`, `hoverHeader` or `collapsible` mode, what do you think about a single prop instead defining those that defaults to `default`? something like 

```ts
interface BaseProps {
mode?: 'default' | 'hoverHeader' | 'collapsible';
}
```

`hoverHeaderOffset` can simply be ignored when `default` or `collapsible`.

Reply:
but maybe we want to avoid a breaking change, which would be required to do this given we'd have to remove (or deprecate) the `hoverHeader` prop. if we want to go with this solution we may also default to `mode= 'hoverHeader'` if `hoverHeader` is `true`.

Reply:
@grafana/dashboards-squad opinions?

Reply:
@Elfo404 I thought about having `mode` but I decided to go with the other option to prevent breaking changes and allow for current implementations to keep using `hoverHeader` prop and not having to deprecate it later. I agree that your proposal is cleaner though.

Reply:
Resolving this since we decided to keep the current solution and not have `mode`.